src/basic/static-destruct.h

```diff
@@ -48,4 +50,7 @@ static inline void static_destruct(void) {
                 d->destroy(d->data);
                 d = ALIGN_TO_PTR(d + 1, __BIGGEST_ALIGNMENT__);
         }
+#else
+        log_debug("Destructors registered with STATIC_DESTRUCTOR_REGISTER have been skipped. See https://github.com/systemd/systemd/pull/10845#issuecomment-443411800.");
```

Comment:
Initially I made it louder but there is a unit test that is supposed to prevent `systemd-something --help` from printing anything to `stderr`. Given that it seems to be the first time anyone has tried to compile `systemd` with `clang` and `asan` I figured nobody is going to read these messages apart from me so I decided that it would be better to downgrade the log message to make the test pass.

Reply:
Because I think the test makes sense (not to just make it green).

Reply:
hmm, i wonder if that unit test shouldn't be updated to either ignore stderr entirely or to set $SYSTEMD_LOG_LEVEL to not be tripped up by LOG_NOTICE errors or so

Reply:
The test uses `SYSTEMD_LOG_LEVEL=info` so `LOG_NOTICE` should suffice too as far as I can tell.

Reply:
I was wrong. LOG_NOTICE is 5 and LOG_INFO is 6 so that won't work. I'm not sure what to do about it. The test looks correct to me as it is.

Reply:
More generally, I'm wondering why anyone would want to always see these warnings running all the commands where _DEFINE_MAIN_FUNCTION is used.